### DISCUSSION TITLE
Question about If activity

### DISCUSSION THREAD

**Discussion by 0xd77bfe** [elsa-workflows/elsa-core#1415] [Q&A]
Hi, I'm trying to understand how `If` activity works.

I've run a simple workflow and I got both `True` and `Done` printed on a console.
`builder.IfTrue(true, whenTrue: a => a.WriteLine("True")).WriteLine("Done");`

Now when I'm trying to run it with false argument, `Done` message is not printed at all😕 
`builder.IfTrue(false, whenTrue: a => a.WriteLine("True")).WriteLine("Done");`

I've tried to specify both `whenTrue` and `whenFalse`.
It works as expected when I setup activities on both paths 
```
builder.If(condition: ctx => false,
    whenTrue: a => a.WriteLine("True"),
    whenFalse: a => a.WriteLine("False"))
 .WriteLine("Done");
```

But when I'm trying to omit whenFalse, then again no `Done` message is visible
```
builder.If(condition: ctx => false,
    whenTrue: a => a.WriteLine("True"),
    whenFalse: a => { })
 .WriteLine("Done");
```

What should I do to continue workflow execution after `If` activity, when condition is not met and I don't want to execute any activities in that case?

**Comment by sfmskywalker** [comment]
I tried to reproduce the issue with the following workflow:

```csharp
public class TestWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .IfTrue(true, a => a.WriteLine("True"))
            .WriteLine("Done");
    }
}
```

When I run this, I see the following output:

```
True
Done
```

As expected.

When I change the condition from `true` to false, I get the following output:

```
Done
```

Also as expected.

It's possible you are using a version of Elsa that contains a bug. Perhaps you can try the latest prerelease (2.3) packages from MyGet to see if you can still reproduce the issue?

**Comment by 0xd77bfe** [comment]
Hi @sfmskywalker 

You're right. It works fine with 2.3.0-preview.93. There must be a bug in 2.2.1
